### ISSUE TITLE
AWS_DEFAULT_REGION in euca2ools not working due to typo of __parse_region(region_envvar)

### ISSUE THREAD

**Issue by castedo** [boto/requestbuilder#6]
I'll describe the user level problem below, but first the line of code I suspect is the root of the problem.

I'm pretty sure the code typo causing the problem I am seeing is due to the following line in boto/requestbuilder, specifically requestbuilder/mixins/__init__.py line 55 in this commit:
https://github.com/boto/requestbuilder/commit/6579df1e4a9392928db746b90f7d95bb6b49c75c.
The line is
`                _user, _region = self.__parse_region(region_envvar)`
and I suspect it should be similar to line 49, that is,
`                _user, _region = self.__parse_region(os.getenv(region_envvar))`

The user level problem I am seeing is with euca2ools 3.3 (more details on installation later).

When I run `euca-describe-instances --region me@us-east-1` I get the expected output. But when I `export AWS_DEFAULT_REGION=me@us-east-1` from the shell and then run `euca-describe-instances` I get:
```
euca-describe-instances: error: No ec2 endpoint to connect to was given. Configured regions with ec2 endpoints are: aws:ap-northeast-1, aws:ap-southeast-1, aws:ap-southeast-2, aws:cn-north-1, aws:eu-central-1, aws:eu-west-1, aws:sa-east-1, aws:us-east-1, aws:us-gov-west-1, aws:us-west-1, aws:us-west-2, localhost
```
When I run `euca-describe-instances --debug` the first lines of output are a good clue to the problem:
```
2016-04-25 11:26:33,399 DEBUG   DescribeInstances parsed arguments: {'filters': None, 'security_token': None, 'InstanceId': [], 'region': None, 'url': None, 'show_empty_fields': False, 'key_id': None, 'secret_key': None}
2016-04-25 11:26:33,399 INFO    Config set region to AWS_DEFAULT_REGION
2016-04-25 11:26:33,399 INFO    Config finding region option user
2016-04-25 11:26:33,399 DEBUG   Config   section AWS_DEFAULT_REGION
2016-04-25 11:26:33,399 DEBUG   Config   section *
2016-04-25 11:26:33,399 INFO    Config   novalue for user
```
Note that the string "AWS_DEFAULT_REGION" is getting parsed and not the content of the shell environment variable, that is: "me@us-east-1".

Regarding the installation, I am on CentOS 7 and I installed euca2ools 3.3 and thus boto requestbuilder 0.5.1 (package python-requestbuilder) by doing
`yum upgrade --enablerepo=epel-testing euca2ools`

**Comment by castedo** [comment]
The documentation mentions environment variable AWS_DEFAULT_REGION here:
http://docs.hpcloud.com/eucalyptus/4.2.1/#shared/environment_variables_precedence.html

**Comment by castedo** [comment]
Rather hacky, but I directly `sudo vim`'d `/usr/lib/python2.7/site-packages/requestbuilder/mixins/__init__.py` :) to make the mentioned one line fix and AWS_DEFAULT_REGION then works.

**Comment by gholms** [comment]
Fixed in commit 4457143, version 0.5.2
